### PR TITLE
Openidauth refresh

### PR DESCRIPTION
**Pull request by sergicastro**

### CODE REVIEW

File: rest/src/main/java/com/abiquo/apiclient/auth/OpenIDConnectAuthentication.java [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+        client.setReadTimeout(requestExecutionContext.getClient().rawClient().getReadTimeout(),
+            TimeUnit.MILLISECONDS);
+        client.setHostnameVerifier(requestExecutionContext.getClient().rawClient()
+            .getHostnameVerifier());
+        client.setSslSocketFactory(requestExecutionContext.getClient().rawClient()
+            .getSslSocketFactory());
+        return client;
+    }
+
+    // TODO: Remove once the api-model-transport dependency is updated to 3.8.6
```
Comment (nacx) [inline rest/src/main/java/com/abiquo/apiclient/auth/OpenIDConnectAuthentication.java]: Can we get rid of this class now?

File: stream/src/main/java/com/abiquo/apiclient/stream/auth/OpenIdAuthentication.java [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+                }
+            }
+        }
+        catch (IOException | InterruptedException | ExecutionException e)
+        {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    // TODO: Remove once the api-model-transport dependency is updated to 3.8.6
```
Comment (nacx) [inline stream/src/main/java/com/abiquo/apiclient/stream/auth/OpenIdAuthentication.java]: Remove this now?

File: stream/src/main/java/com/abiquo/apiclient/stream/StreamClient.java [hunk 3]
```diff
@@ -165,7 +178,40 @@ public void connect() throws IOException
...
+
+    private boolean isReconnectConfigured()
+    {
+        return !manuallyClosing.get() && reconnect;
+    }
+
+    /**
+     * A {@link org.atmosphere.wasync.Function} that processes the stream received events.
+     */
+    private Function<String> onMessage()
```
Comment (nacx) [inline stream/src/main/java/com/abiquo/apiclient/stream/StreamClient.java]: Make these functions variables instead of methods?

File: stream/src/main/java/com/abiquo/apiclient/stream/auth/Authentication.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ */
+package com.abiquo.apiclient.stream.auth;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public abstract class Authentication
+{
+    public abstract String build();
+
+    public void validateTokens(final ObjectMapper json)
```
Comment (nacx) [inline stream/src/main/java/com/abiquo/apiclient/stream/auth/Authentication.java]: I'd try to get rid of this in the Authentication class. Having an ObjectMapper as a parameter looks completely unrelated to Authentication; it's more an implementation detail that should remain only in the subclasses where it makes sense.

**Comment by sergicastro** [comment]
:up: 
Don't remember if I miss something about we talk. Let me know.
